js/filter.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+window.filter = (function () {
+  var filterContainer = document.querySelector('.map__filters-container');
+  var filterTypeElement = filterContainer.querySelector('#housing-type');
+  var filterPriceElement = filterContainer.querySelector('#housing-price');
+  var filterRoomsElement = filterContainer.querySelector('#housing-rooms');
+  var filterGuestsElement = filterContainer.querySelector('#housing-guests');
+  var filterFeaturesElement = filterContainer.querySelector('#housing-features');
+  var housingFeaturesCheckboxes = document.querySelector('#housing-features').querySelectorAll('.map__checkbox');
+
+  function filterAds(unfilteredData) {
+    var type = filterTypeElement.value;
+    var price = filterPriceElement.value;
+    var rooms = filterRoomsElement.value;
+    var guests = filterGuestsElement.value;
+    var featuresArray = [];
+
+    Array.prototype.forEach.call(housingFeaturesCheckboxes, function (elem) {
+      if (elem.checked) {
+        featuresArray.push(elem.value);
+      }
+    });
+
+    function isCorrectType(ad) {
+      if (type !== 'any') {
+        return ad.offer.type === type;
+      }
+      return ad;
+    }
+
+    function isCorrectPrice(ad) {
+      if (price !== 'any') {
+        var returnProperty = {
```

Comment:
define it outside of the function in a separate variable. why to create it every time.

Reply:
done